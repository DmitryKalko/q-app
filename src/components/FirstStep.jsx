import React from "react";

import data from "../store/data";
import { observer } from "mobx-react-lite";

const FirstStep = observer(({ toNextStep, lightTheme }) => {
  const setFirstName = (e) => {
    data.setFirstName(e.target.value);
  };
  const setLastName = (e) => {
    data.setLastName(e.target.value);
  };

  return (
    <div className="first-step">
      <input
        type={"text"}
        placeholder="first name"
        style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
        onChange={setFirstName}
        value={data.store.firstName}
      />
      <input
        type={"text"}
        placeholder="second name"
        style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
        onChange={setLastName}
        value={data.store.lastName}
      />
      <button
        className="btn"
        onClick={toNextStep}
        style={lightTheme ? { color: "white" } : { color: "#6d6d6d" }}
      >
        NEXT
      </button>
    </div>
  );
});

export default FirstStep;
