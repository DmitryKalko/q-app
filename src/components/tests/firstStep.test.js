import { shallow } from "enzyme";
import React from "react";
import FirstStep from "../FirstStep";


const setUp = () => shallow(<FirstStep  />);

describe('FirstStep component', () => {
    let component;
    let instance;

    beforeEach(() => {
        component = setUp();  // сюда присваивается результат функции shallow, которая выполняет отрисовку компонента
        instance = component.instance(); 
        // у результата работы shallow есть метод instance, который возвращает наш компонент. после вызова instance компонента мы можем обращаться к его свойствам и методам
    })
    test('should render FirstStep component', () => {
        expect(component).toMatchSnapshot();
    });


    it('FirstStep inputs', () => {
        const testState = { name: 'denn', lastName: 'kalko' };
        expect(component.find('input').at(0).prop('value')).toEqual('');
        expect(component.find('input').at(1).prop('value')).toEqual('');
        component.find('input').at(0).simulate('change', { target: { name: 'name', value: 'denn' } });
        component.find('input').at(1).simulate('change', { target: { name: 'age', lastName: 'kalko' } });
        expect(testState.name).toEqual('denn');
        expect(testState.lastName).toEqual('kalko');
      });

})