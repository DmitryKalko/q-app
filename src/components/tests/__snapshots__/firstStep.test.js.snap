// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirstStep component should render FirstStep component 1`] = `
<div
  className="first-step"
>
  <input
    onChange={[Function]}
    placeholder="first name"
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
    type="text"
    value=""
  />
  <input
    onChange={[Function]}
    placeholder="second name"
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
    type="text"
    value=""
  />
  <button
    className="btn"
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
  >
    NEXT
  </button>
</div>
`;

exports[`FirstStep component should render component with props for dark theme 1`] = `
<div
  className="first-step"
>
  <input
    onChange={[Function]}
    placeholder="first name"
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
    type="text"
    value="denn"
  />
  <input
    onChange={[Function]}
    placeholder="second name"
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
    type="text"
  />
  <button
    className="btn"
    style={
      Object {
        "color": "#6d6d6d",
      }
    }
  >
    NEXT
  </button>
</div>
`;

exports[`FirstStep component should render component with props for light theme 1`] = `
<div
  className="first-step"
>
  <input
    onChange={[Function]}
    placeholder="first name"
    style={
      Object {
        "color": "white",
      }
    }
    type="text"
    value="denn"
  />
  <input
    onChange={[Function]}
    placeholder="second name"
    style={
      Object {
        "color": "white",
      }
    }
    type="text"
  />
  <button
    className="btn"
    style={
      Object {
        "color": "white",
      }
    }
  >
    NEXT
  </button>
</div>
`;
